### PR TITLE
wlr-foreign-toplevel-management: misc fixes/improvements

### PR DESCRIPTION
**Pull request by ifreund**
Just a couple of things I noticed while working on the new foreign-toplevel protocols.

### CODE REVIEW

File: types/wlr_foreign_toplevel_management_v1.c [hunk 1]
```diff
@@ -195,6 +202,10 @@ void wlr_foreign_toplevel_handle_v1_set_title(
 		struct wlr_foreign_toplevel_handle_v1 *toplevel, const char *title) {
 	free(toplevel->title);
 	toplevel->title = strdup(title);
+	if (toplevel->title) {
```
[the remark this answers is not in the document]
Comment (ifreund) [reply types/wlr_foreign_toplevel_management_v1.c:205]: Of course, thanks for catching that!

File: types/wlr_foreign_toplevel_management_v1.c [hunk 2]
```diff
@@ -127,13 +130,17 @@ static void foreign_toplevel_handle_close(struct wl_client *client,
...
-		struct wl_resource *resource, struct wl_resource *surface,
+		struct wl_resource *resource, struct wl_resource *surface_resource,
 		int32_t x, int32_t y, int32_t width, int32_t height) {
 	struct wlr_foreign_toplevel_handle_v1 *toplevel =
 		toplevel_handle_from_resource(resource);
 	if (!toplevel) {
 		return;
 	}
+	struct wlr_surface *surface = wlr_surface_from_resource(surface_resource);
+	if (!surface) {
```
Comment (emersion) [inline types/wlr_foreign_toplevel_management_v1.c:141]: I don't think a surface can be inert.

Comment (ifreund) [reply types/wlr_foreign_toplevel_management_v1.c:141]: Looks like you're right. I skimmed all the `wl_resource_set_user_data()` calls in wlroots but didn't find any setting the `wl_surface` resource's user data to `NULL`.

Comment (ifreund) [reply types/wlr_foreign_toplevel_management_v1.c]: #2597

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Review by ammen99 — APPROVED** [review]
Thanks :)

**Pull request merged**